pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -947,3 +953,23 @@ func GetBackupFilename(backup *v3.EtcdBackup) string {
 	}
 	return snapshot
 }
+
+// transform an imported cluster into a k3s cluster using its discovered version
+func k3sClusterConfig(cluster *v3.Cluster) error {
+	// version is not found until cluster is provisioned
+	if cluster.Status.Driver == "" || cluster.Status.Version == nil {
+		return &controller.ForgetError{
+			Err:    fmt.Errorf("waiting for full cluster configuration"),
+			Reason: "Pending"}
+	}
+	// TODO Rancher may support upgrading the local cluster if its k3s in the future
```

Comment:
yeah, good call out here. In fact, I think we will want to support this out of the gate. You don't have to do it in this PR (because I really want to get the baseline functionality merged for wes), but I think you'll have to work through the usecase of flipping the local cluster to be k3s when appropriate. I looked through the code and switching the cluster driver seems like it will be fine.